blocks/snippets.py

```diff
@@ -0,0 +1,14 @@
+"""Routines to build frequently used Theano expressions."""
+from theano import tensor
+
+
+def L2_norm(tensors):
+    """Computes the total L2 norm of a set of tensors.
+
+    Parameters
+    ----------
+    tensors : iterable of :class:`~theano.Variable`
+        The tensors.
+
+    """
+    return tensor.sqrt(sum((t ** 2).sum() for t in tensors))
```

Comment:
`theano_expressions` then? The thing is that a "snippet" is almost always used to refer to a piece of code that you copy and paste, which is the opposite of what you're doing here.

I don't think that using `tensor.sqr` is in any way confusing though. It's a simple element-wise operator (the name of which describes exactly what it does), which takes multiple inputs, like all element-wise operators in Theano. To me that line is actually much easier to read and shorter than using a generator within a function. I'd also be surprised if the documentation of `tensor.sqr` is longer than 10 words...

And you'll probably say this is negligible, which might be true, but it also makes for a much smaller computational graph than summing in two sweeps instead of one, which might help save on compilation time (assuming that Theano optimizes your expression correctly).

```
Elemwise{sqrt,no_inplace} [@A] ''
 |Sum{acc_dtype=float64} [@B] ''
   |Elemwise{sqr,no_inplace} [@C] ''
     |Join [@D] ''
       |TensorConstant{0} [@E]
       |DimShuffle{x,0} [@F] ''
       | |<TensorType(float32, vector)> [@G]
       |DimShuffle{x,0} [@H] ''
       | |<TensorType(float32, vector)> [@I]
       |DimShuffle{x,0} [@J] ''
       | |<TensorType(float32, vector)> [@K]
       |DimShuffle{x,0} [@L] ''
       | |<TensorType(float32, vector)> [@M]
       |DimShuffle{x,0} [@N] ''
         |<TensorType(float32, vector)> [@O]
```
While the current one gives
```
Elemwise{sqrt,no_inplace} [@A] ''
 |Elemwise{add,no_inplace} [@B] ''
   |Elemwise{add,no_inplace} [@C] ''
   | |Elemwise{add,no_inplace} [@D] ''
   | | |Elemwise{add,no_inplace} [@E] ''
   | | | |Elemwise{add,no_inplace} [@F] ''
   | | | | |TensorConstant{0} [@G]
   | | | | |Sum{acc_dtype=float64} [@H] ''
   | | | |   |Elemwise{pow,no_inplace} [@I] ''
   | | | |     |<TensorType(float32, vector)> [@J]
   | | | |     |DimShuffle{x} [@K] ''
   | | | |       |TensorConstant{2} [@L]
   | | | |Sum{acc_dtype=float64} [@M] ''
   | | |   |Elemwise{pow,no_inplace} [@N] ''
   | | |     |<TensorType(float32, vector)> [@O]
   | | |     |DimShuffle{x} [@P] ''
   | | |       |TensorConstant{2} [@L]
   | | |Sum{acc_dtype=float64} [@Q] ''
   | |   |Elemwise{pow,no_inplace} [@R] ''
   | |     |<TensorType(float32, vector)> [@S]
   | |     |DimShuffle{x} [@T] ''
   | |       |TensorConstant{2} [@L]
   | |Sum{acc_dtype=float64} [@U] ''
   |   |Elemwise{pow,no_inplace} [@V] ''
   |     |<TensorType(float32, vector)> [@W]
   |     |DimShuffle{x} [@X] ''
   |       |TensorConstant{2} [@L]
   |Sum{acc_dtype=float64} [@Y] ''
     |Elemwise{pow,no_inplace} [@Z] ''
       |<TensorType(float32, vector)> [@BA]
       |DimShuffle{x} [@BB] ''
         |TensorConstant{2} [@L]
```